### PR TITLE
Document more of the codegen features

### PR DESCRIPTION
**Pull request by axic**

### CODE REVIEW

File: libsolidity/codegen/CompilerContext.h [hunk 1]
```diff
@@ -56,7 +56,9 @@ class CompilerContext
 			m_runtimeSub = size_t(m_asm->newSub(m_runtimeContext->m_asm).data());
 	}
 
+	/// Update currently enabled set of experimental features.
 	void setExperimentalFeatures(std::set<ExperimentalFeature> const& _features) { m_experimentalFeatures = _features; }
+	/// @returns true if the given feature is enabled.
```
Comment (chriseth) [inline libsolidity/codegen/CompilerContext.h:61]: Which of the source units does it check?

Comment (axic) [reply libsolidity/codegen/CompilerContext.h:61]: The above supplied list.

**Review by chriseth — APPROVED** [review]

File: libsolidity/codegen/CompilerUtils.h [hunk 2]
```diff
@@ -38,14 +38,20 @@ class CompilerUtils
...
+	/// Stack pre: <mem_end>
+	/// Stack post:
 	void storeFreeMemoryPointer();
 	/// Allocates a number of bytes in memory as given on the stack.
 	/// Stack pre: <size>
 	/// Stack post: <mem_start>
 	void allocateMemory();
 	/// Appends code that transforms memptr to (memptr - free_memptr) memptr
+	/// Stack pre: <mem_end>
+	/// Stack post: <size> <mem_start>
```
Comment (axic) [inline libsolidity/codegen/CompilerUtils.h:54]: This seems weird but that's how the code seems to be.

File: libsolidity/codegen/CompilerContext.h [hunk 3]
```diff
@@ -78,13 +80,15 @@ class CompilerContext
...
 	eth::AssemblyItem functionEntryLabelIfExists(Declaration const& _declaration) const;
-	void setInheritanceHierarchy(std::vector<ContractDefinition const*> const& _hierarchy) { m_inheritanceHierarchy = _hierarchy; }
 	/// @returns the entry label of the given function and takes overrides into account.
 	FunctionDefinition const& resolveVirtualFunction(FunctionDefinition const& _function);
 	/// @returns the function that overrides the given declaration from the most derived class just
 	/// above _base in the current inheritance hierarchy.
 	FunctionDefinition const& superFunction(FunctionDefinition const& _function, ContractDefinition const& _base);
+	/// @returns the next constructor in the inheritance hierarchy.
 	FunctionDefinition const* nextConstructor(ContractDefinition const& _contract) const;
+	/// Sets the current inheritance hierarchy from derived to base.
```
Comment (axic) [inline libsolidity/codegen/CompilerContext.h:90]: Haven't verified this, basing it on `superFunction`.

File: libsolidity/codegen/CompilerContext.h (hunk 1 above)
Comment (chriseth) [reply libsolidity/codegen/CompilerContext.h:61]: Ah right :-)

File: libsolidity/codegen/CompilerContext.h (hunk 3 above)
Comment (chriseth) [reply libsolidity/codegen/CompilerContext.h:90]: looks good.

File: libsolidity/codegen/CompilerUtils.h (hunk 2 above)
Comment (chriseth) [reply libsolidity/codegen/CompilerUtils.h:54]: This is handy for many EVM instructions.

**Review by chriseth — APPROVED** [review]

**Pull request merged**